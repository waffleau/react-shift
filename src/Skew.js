import React, {PropTypes, PureComponent} from 'react'

import Transition from './Transition'

export class Skew extends PureComponent {
  static defaultProps = {
    easing: 'easeInOut',
    from: [0, 0]
  }

  static propTypes = {
    ...Transition.baseProps,
    from: PropTypes.arrayOf(PropTypes.number),
    to: PropTypes.arrayOf(PropTypes.number).isRequired
  }

  getFromValue() {
    const {from} = this.props
    return {
      transform: `skew(${from[0]}deg, ${from[1]}deg)`
    }
  }

  getStyle() {
    const {style} = this.props
    return {
      transformOrigin: 'center center',
      ...style
    }
  }

  getToValue() {
    const {to} = this.props
    return {
      transform: `skew(${to[0]}deg, ${to[1]}deg)`
    }
  }

  render() {
    const {children} = this.props
    return (
      <Transition
        {...this.props}
        style={this.getStyle()}
        fromValue={this.getFromValue()}
        toValue={this.getToValue()}>
        {children}
      </Transition>
    )
  }
}

export default Skew
