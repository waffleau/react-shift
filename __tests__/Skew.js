import React from 'react'
import renderer from 'react-test-renderer'

import Skew from '../src/Skew'

test('has no scale when inactive', () => {
  const tree = renderer.create(
    <Skew
      active={false}
      to={[10, 10]}>
      {'Children'}
    </Skew>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a minimum x skew', () => {
  const tree = renderer.create(
    <Skew
      active={false}
      from={[.1, .1]}
      to={[10, 10]}>
      {'Children'}
    </Skew>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a minimum y skey', () => {
  const tree = renderer.create(
    <Skew
      active={false}
      from={[.1, .1]}
      to={[10, 10]}>
      {'Children'}
    </Skew>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a maximum x skew', () => {
  const tree = renderer.create(
    <Skew
      active
      to={[1.1, 1.1]}>
      {'Children'}
    </Skew>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a maximum y skew', () => {
  const tree = renderer.create(
    <Skew
      active
      to={[10, 10]}>
      {'Children'}
    </Skew>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
