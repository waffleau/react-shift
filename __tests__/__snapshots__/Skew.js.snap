exports[`test can specify a maximum x skew 1`] = `
<div
  style={
    Object {
      "transform": "skew(1.1deg, 1.1deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can specify a maximum y skew 1`] = `
<div
  style={
    Object {
      "transform": "skew(10deg, 10deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can specify a minimum x skew 1`] = `
<div
  style={
    Object {
      "transform": "skew(0.1deg, 0.1deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can specify a minimum y skey 1`] = `
<div
  style={
    Object {
      "transform": "skew(0.1deg, 0.1deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test has no scale when inactive 1`] = `
<div
  style={
    Object {
      "transform": "skew(0deg, 0deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;
